pkg/controller/deployment/deployment_controller.go

```diff
@@ -133,15 +133,15 @@ func NewDeploymentController(dInformer extensionsinformers.DeploymentInformer, r
 		DeleteFunc: dc.deletePod,
 	})
 
-	dc.syncHandler = dc.syncDeployment
-	dc.enqueueDeployment = dc.enqueue
-
 	dc.dLister = dInformer.Lister()
 	dc.rsLister = rsInformer.Lister()
 	dc.podLister = podInformer.Lister()
 	dc.dListerSynced = dInformer.Informer().HasSynced
 	dc.rsListerSynced = rsInformer.Informer().HasSynced
 	dc.podListerSynced = podInformer.Informer().HasSynced
```

Comment:
Looks like this is one of the bits of work that got lost when https://github.com/kubernetes/kubernetes/pull/51492 was closed. As a defensive maneuver, inline everything possible into the DeploymentController instance creation.

Reply:
And anything that's not inlined, move to lines preceding calls to `AddEventHandler`.